docs/static/persistent-queues.asciidoc

```diff
@@ -0,0 +1,127 @@
+queue and processes the events in the backlog. 
+
+If Logstash crashes or experiences an uncontrolled shutdown, any in-flight
+events are left as unACKed in the persistent queue. Upon restart, Logstash will
+replay the events from its history, potentially leading to duplicate data being
+written to the output.
+
+[[configuring-persistent-queues]]
+==== Configuring Persistent Queues
+
+To configure persistent queues, you can specify the following options in the
+Logstash settings file:
+
+* `queue.type`: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (`queue.type: memory`).
+* `path.queue`: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
+* `queue.page_capacity`: The size of the page data file. The queue data consists of append-only data files separated into pages. The default size is 250mb. 
+* `queue.max_events`:  The maximum number of unread events that are allowed in the queue. The default is 0 (unlimited).
+* `queue.max_bytes`: The total capacity of the queue in number of bytes. The
+default is 1024mb (1g). Make sure the capacity of your disk drive is greater
+than the value you specify here. If both `queue.max_events` and 
+`queue.max_bytes` are specified, Logstash uses whichever criteria is reached
+first. 
+
+You can also specify options that control when the checkpoint file gets updated (`queue.checkpoint.acks`, `queue.checkpoint.writes`, and
+`queue.checkpoint.interval`). See <<logstash-settings-file>> for more
+information about these options.
+
+Example configuration:
+
+[source, yaml]
+queue.type: persisted
+queue.max_bytes: 4g 
+
+[[backpressure-persistent-queues]]
+==== Handling Back Pressure
+
+Logstash has a built-in mechanism that adds back pressure to the data flow when
+the queue is full. This mechanism helps Logstash control the rate of data flow
+at the input stage without overwhelming downstream stages and outputs like
+Elasticsearch.
+
+You can control when back pressure happens by using the `queue.max_bytes` 
+setting to configure the capacity of the queue on disk. The following example
+sets the total capacity of the queue to 8gb:
+
+[source, yaml]
+queue.type: persisted
+queue.max_bytes: 8gb
+
+With these settings specified, Logstash will buffer unACKed events on disk until 
+the size of the queue reaches 8gb. When the queue is full of unACKed events, and
+the size limit has been reached, Logstash will no longer accept new events. This
+process is called back pressure.
+
+Each input handles back pressure independently. For example, when the
+<<plugins-inputs-beats,beats>> input encounters back pressure, it no longer
+accepts new connections and waits until the persistent queue has space to accept
+more events. After the filter and output stages finish processing existing
+events in the queue and ACKs them, Logstash automatically starts accepting new
+events.
```

Comment:
I think we should also have a short section on monitoring PQs. This can mostly just link to the monitoring API queue stats docs.